src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -97,6 +103,38 @@ uct_cuda_copy_get_mem_type(uct_md_h md, void *address, size_t length)
     return mem_info.type;
 }
 
+static inline
+ucs_status_t uct_cuda_copy_get_ctx_rsc(uct_cuda_copy_iface_t *iface,
+                                       uct_cuda_copy_per_ctx_rsc_t **ctx_rsc)
+{
+    CUcontext current_ctx;
+    ucs_status_t status;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&current_ctx));
+    if (status != UCS_OK) {
+        return status;
+    } else if (current_ctx == NULL) {
+        ucs_error("attempt to perform cuda memcpy without active context");
```

Comment:
I think this log message can be confusing since the function can be called by different callers. It is better to print the following message: `there is no cuda context bound to the calling cpu thread`.